src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PolicyFunctionsTest.kt

```diff
@@ -0,0 +1,132 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.AdditionalCondition
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.AdditionalConditionData
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.AdditionalConditionAp
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.ConditionChangeType.*
+
+class PolicyFunctionsTest {
+
+  @Nested
+  inner class `Condition changes` {
+    @Test
+    fun `no changes when conditions are empty`() {
+      assertThat(licencePolicyChanges(emptyList(), emptyList(), emptyMap())).isEmpty()
+    }
+
+    @Test
+    fun `no changes when single condition with no changes`() {
+      val condition = additionalCondition().copy(code = "code-1", text = "some-text")
+      val policyCondition = policyCondition().copy(code = "code-1", text = "some-text")
+
+      assertThat(licencePolicyChanges(listOf(condition), listOf(policyCondition), emptyMap())).isEmpty()
+    }
+
+    @Test
+    fun `no changes when new condition added`() {
+      val condition = additionalCondition().copy(code = "code-1", text = "some-text")
+      val matchingPolicyCondition = policyCondition().copy(code = "code-1", text = "some-text")
+      val policyCondition = policyCondition().copy(code = "code-2", text = "some-text")
+
+      assertThat(
+        licencePolicyChanges(
+          listOf(condition),
+          listOf(matchingPolicyCondition, policyCondition),
+          emptyMap()
+        )
+      ).isEmpty()
+    }
+
+    @Test
+    fun `no changes when same placeholders`() {
+      val placeHolders = mapOf("code-1" to mapOf("SOME CHANGE" to "ph-1"))
+      val condition = additionalCondition().copy(
+        code = "code-1", text = "some-text", data = listOf(
+          AdditionalConditionData(field = "ph-1")
+        )
+      )
+      val policyCondition = policyCondition().copy(code = "code-1", text = "some-text")
+
+      assertThat(licencePolicyChanges(listOf(condition), listOf(policyCondition), placeHolders)).isEmpty()
+    }
+
+    @Disabled("Should pass?")
+    @Test
```

Comment:
is this a bug? 

when the stored condition has no placeholders but the new policy version has some, it should return it as a change?

Reply:
For visibility:
We have discussed this on a call and the outcome was that we believe that this is correct expected behaviour, but will need to confirm. As there don't seem to be any changes of this type in the current policy changes (and with the imminent deadline), we're going to progress with the current behaviour and consider this post-release.